AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -169,6 +171,19 @@ - (void)setDelegate:(id<ASNetworkImageNodeDelegate>)delegate
 - (void)displayWillStart
 {
   [super displayWillStart];
+  
+  if (_cacheSupportsSynchronousFetch) {
+    {
+      ASDN::MutexLocker l(_lock);
+      if (_URL && _downloadIdentifier == nil) {
+        UIImage *result = [_cache synchronouslyFetchedCachedImageWithURL:_URL];
```

Comment:
Just a sanity check for my understanding, is it guaranteed that this image will always be compressed? Based on our configuration, I assume that is the case, but just checking